server/blob/blob-union/src/main/java/org/apache/james/blob/union/UnionBlobStore.java

```diff
@@ -141,38 +141,38 @@ boolean streamHasContent(PushbackInputStream pushBackIS) throws IOException {
         return false;
     }
 
-    private CompletableFuture<byte[]> readBytesFallBackIfFailsOrEmptyResult(BlobId blobId) {
+    private Mono<byte[]> readBytesFallBackIfFailsOrEmptyResult(BlobId blobId) {
         return currentBlobStore.readBytes(blobId)
-            .thenApply(Optional::ofNullable)
-            .exceptionally(this::logAndReturnEmptyOptional)
-            .thenCompose(maybeBytes -> readFromLegacyIfNeeded(maybeBytes, blobId));
+            .map(Optional::ofNullable)
```

Comment:
I remember before that `Optional::ofNullable` is for dealing with poor `CompletableFuture` composability - `CompletableFuture` doesn't have `doOnError(Consumer<>)` but only `exceptionally(Function<Throwable, T>)` where it require a function to generate result on error, and using Optional for cheating with `CompletableFuture`.

We may don't need that workaround when using reactor

Reply:
not related but ok, I added a commit about that.